### ISSUE TITLE
[jvm-packages] -Ve values from multi:softmax objective.

### ISSUE THREAD

**Issue by tanwanirahul** [dmlc/xgboost#1293]
Hi Devs.

I was running a multi class classification on the sample data attached. Model is trained just fine, however when I use the model for predictions, I see negative values in the output.

**Code:**

`import ml.dmlc.xgboost4j.scala.{Booster, DMatrix}
import ml.dmlc.xgboost4j.scala.spark.{DataUtils, XGBoost}
import org.apache.spark.{SparkConf, SparkContext}
import org.apache.spark.mllib.util.MLUtils

import DataUtils._

val trainRDD = MLUtils.loadLibSVMFile(sc, "data/mllib/sample_binary_classification_data.txt")
val testRDD = trainRDD.sample(false, 0.1)

val paramMap = List(
            "max_depth" -> 6,
            "num_class" -> 3,
            "booster" -> "gbtree",
            "objective" -> "multi:softmax").toMap

val xgboostModel = XGBoost.train(trainRDD, paramMap, 20, nWorkers = 4, useExternalMemory = false)

val output = xgboostModel.predict(new DMatrix(testRDD.collect().iterator))
output.foreach { arr => println(); arr.foreach { x => print(x + " ") } }`

**Output:**

`
-1.7982326 3.5512266 -2.008376

-1.627734 3.411692 -2.008376 

3.4890263 -1.6983604 -2.008376 

-1.8316703 3.6790173 -1.2907562 

3.3072288 -1.4880885 -1.0675741 

-1.8316703 3.6790173 -2.1104112 

-1.8316703 2.9718113 -1.9851551 

-1.9200544 3.5512266 -2.1104112 

3.4890263 -1.5947924 -2.1104112
`

**Data:**

I am using the standard binary_classification dataset that is shipped with spark with a small tweak. I was just changing the first two labels to 2 to make it 3 class classifier. Attached is the training file.
[sample_binary_classification_data.txt](https://github.com/dmlc/xgboost/files/327727/sample_binary_classification_data.txt)

I may be doing something wrong with the parameters being passed; but couldn't figure out based on the information available to me on console / logs / official documentation.

I was then explicitly setting rest of the parameters to default values with the following map, but still the same scores.

`
val paramMap = List(
            "max_depth" -> 6,
            "num_class" -> 3,
            "gamma" -> 0,
            "eta" -> 0.3,
            "eval_metric" -> "merror",
            "booster" -> "gbtree",
            "min_child_weight" -> 1,
            "max_delta_step" -> 0,
            "subsample" -> 1,
            "colsample_bytree" -> 1,
            "seed" -> 0,
            "base_score" -> 0.5,
            "objective" -> "multi:softmax").toMap
`

thanks!

**Comment by tanwanirahul** [comment]
@CodingCat I ran the XGboost implementation for SciKit with the same data and parameters and probability values were correct.

**Comment by CodingCat** [comment]
it might be caused by the missing values, would you try https://github.com/dmlc/xgboost/blob/master/jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/DMatrix.java#L128 to construct DMatrix?

**Comment by tanwanirahul** [comment]
@CodingCat Sure, I will give it a try. Also, I will try the same code with the dense matrix.
